compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -418,15 +399,35 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
                 false
               }
             }
-            .map(_._2)
+            .map(pair => (pair._2, pair._1.pos))
 
           regularClasses ::: staticForwarderClasses
         }
 
-        generatedJSAST(clDefs)
+        generatedJSAST(clDefs.map(_._1))
```

Comment:
This is not ideal IMO:
- We create an intermediate list for a hook that is only used for testing.
- We don't actually provide the final tree to the test.

Maybe we can change the contract so this can be called multiple times and just call it in the loop below?

Reply:
Good point. I added a commit that changes the contract.